keras/layers/normalization/batch_normalization.py

```diff
@@ -282,3 +287,44 @@ def get_config(self):
             "synchronized": self.synchronized,
         }
         return {**base_config, **config}
+
+    def _moments(self, inputs, mask):
+        if mask is None:
+            return ops.moments(
+                inputs,
+                axes=self._reduction_axes,
+                synchronized=self.synchronized,
+            )
+
+        mask_weights = ops.cast(
+            mask,
+            inputs.dtype,
+        )
+        mask_weights_broadcasted = ops.expand_dims(
+            mask_weights,
+            axis=-1,
+        )
+        weighted_inputs = mask_weights_broadcasted * inputs
+
+        weighted_input_sum = ops.sum(
+            weighted_inputs,
+            self._reduction_axes,
+            keepdims=True,
+        )
+        sum_of_weights = ops.sum(
+            mask_weights_broadcasted,
+            self._reduction_axes,
+            keepdims=True,
+        )
+        mean = weighted_input_sum / (sum_of_weights + self.epsilon)
```

Comment:
An issue here is that `self.epsilon` is the batch norm epsilon, used to compute the inverse of the variance in the `batch_normalization` op. It is usually set to a fairly large value -- for instance, the default value is 1e-3, and some folks use 1e-2. So the moment computation here will be very imprecise.

Prefer using `config.epsilon()`, which is typically 1e-7.

Reply:
That makes sense. Fixing, thanks!